modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/internal/ConnectionKey.java

```diff
@@ -25,6 +25,9 @@
  * Connection Key.
  */
 public class ConnectionKey {
+    /** */
+    private final Object consistentId;
```

Comment:
`consistentId` should be a part of session meta (`ses.meta(CONSISTENT_ID_META)`. It has no meaning for the connectionKey class. An thus it will require much less changes in this PR.

Reply:
I don't think that this would require less changes, pretty much the same amount.
I don't see any issues in storing consistentId near to nodeId, looks logical to me. Why do you think that it has no meaning?

Reply:
I see the following major reasons:
1. ConnectionKey is a public class from **public API**. Do you think changing public API is a good idea for a minor bug?
1. `consistentId` doesn't participate in `equals` and `hashCode` methods. For the person, who will take a look at ConnectionKey class the first time this would look weird because `consistentId` is not the same as `nodeId`.
2. This field is used only for metrics and does not have any meaning for a session connection key itself.
3. You are passing `consistentId` key for each constructor rather obtaining it from session attributes directly, so you should change each constructor and this will lead to unnecessary changes.

Reply:
1. No, it's not: org.apache.ignite.spi.communication.tcp.**internal**.ConnectionKey
2. It's absolutely fine, other fields don't participate in equals and hashCode either.

Fine, I'll do it and we'll see if it got simpler. Maybe it is, I don't know for sure.

Reply:
just for the protocol - I disagree with this change)

Reply:
@ibessonov thank you, please, let me know when you'll be done with changes. I'll take a final look.